### PR TITLE
move LoadLibrary logic to new win_util module

### PR DESCRIPTION
**Pull request by ajax16384**
use get_library_procedure as common windows dyn proc load routine

### CODE REVIEW

File: include/libtorrent/win_util.hpp [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+	struct kernel32 {
+		static constexpr char const* library_name = "kernel32.dll";
+	};
+
+	struct advapi32 {
+		static constexpr char const* library_name = "advapi32.dll";
+	};
+
+} // namespace libtorrent
```
Comment (arvidn) [inline include/libtorrent/win_util.hpp]: sorry. would you mind moving this file to include/libtorrent/aux_/win_util.hpp? and also put the functions in namespace libtorrent::aux

**Comment by arvidn** [comment]
please also add the header file to include/libtorrent/Makefile.am
otherwise, lgtm

File: include/libtorrent/win_util.hpp (hunk 1 above)
Comment (ajax16384) [inline include/libtorrent/win_util.hpp]: ok

**Comment by arvidn** [comment]
lgtm

**Pull request merged**